internal/docker/executor.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 deploy-agent authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package docker
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fsouza/go-dockerclient"
+	"github.com/tsuru/tsuru/exec"
+)
+
+// Executor uses docker exec to execute a command in a running docker container
+type Executor struct {
+	ContainerID string
+	Client      *Client
+	DefaultUser string
+}
+
+func (d *Executor) Execute(opts exec.ExecuteOptions) error {
+	return d.ExecuteAsUser(d.DefaultUser, opts)
+}
+
+func (d *Executor) ExecuteAsUser(user string, opts exec.ExecuteOptions) error {
+	cmd := append([]string{opts.Cmd}, opts.Args...)
+	if opts.Dir != "" {
+		cmd = append(cmd[:2], fmt.Sprintf("cd %s && %s", opts.Dir, strings.Join(cmd[2:], " ")))
```

Comment:
I'm a bit worried here by two things here. First, the `cmd[:2]` seems to assume that cmd always start with `["bash", "-c"]`, I think this is the case for now but I don't know if this is safe in the future.

Second, the `strings.Join(cmd[n:], " ")` part, if some arguments include spaces this will break, we would have to shell quote each arg to ensure this doesn't happen. I'm thinking something like `cmd == []string{"rm", "my file"}` after this line it could become `cmd == []string{"cd /dir && rm my file"}` (if bash -c is not assumed to be present).